### PR TITLE
Feature/use account lib

### PR DESCRIPTION
**Pull request by romeovs**
Removes the local version of `core/account` in favor of https://github.com/TheThingsNetwork/go-account-lib

Notable changes:
- `tokenkey` provider is now a part of `go-account-lib` and supports different caching strategies
- `ttnctl` now stores tokens and derived tokens in the directory specified by `--token-dir` (default `$HOME/.ttnctl`) instead of in the config file, it names the file after which account server the token is for.

### CODE REVIEW

File: core/discovery/server.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (d *discoveryServer) checkMetadataEditRights(ctx context.Context, in *pb.Me
 		}
 		// Allow APP_ID announcements from all trusted auth servers
 		// When announcing APP_ID, token is user token that contains apps
-		if !claims.CanEditApp(string(in.Metadata.Value)) {
+		if !claims.AppRight(string(in.Metadata.Value), "settings") {
```
Comment (johanstokking) [inline core/discovery/server.go]: Create a `const` for this

**Comment by romeovs** [comment]
Do not merge yet, I'm resolving an issue with the handler manager in `ttnctl`

**Comment by romeovs** [comment]
This should be ready for a merge

File: ttnctl/cmd/applications_pf.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ function Decoder(bytes) {
 
 		appID := util.GetAppID(ctx)
 
-		conn, manager := util.GetHandlerManager(ctx)
+		conn, manager := util.GetHandlerManager(ctx, appID)
 		defer conn.Close()
 
 		app, err := manager.GetApplication(appID)
```
Comment (htdvisser) [inline ttnctl/cmd/applications_pf.go]: Why do we need to pass `appID` to `manager.GetApplication` when `manager` is already scoped to this `AppID`? (same for other `ttnctl` cmds)

Comment (romeovs) [inline ttnctl/cmd/applications_pf.go]: The manager does not actually know what appID its token is for. We could refactor it to know though.

Comment (romeovs) [inline ttnctl/cmd/applications_pf.go]: Basically:
```
util.GetHandlerManager(ctx, appID)
```
Gets a handler manager that is preloaded with the correct token.
```
app, err := manager.GetApplication(appID)
```
Fetches the application.

**Pull request merged**